_additional/frontend/front.js

```diff
@@ -0,0 +1,130 @@
+var data = {};
+            var users = {};
+            var name = '';
+            var currentMessages = new Set();
+            var currentUsers = new Set();
+
+            function getUsers () {
+                $.ajax({
+                    type: 'GET',
+                    url: '/users',
+                    dataType: "JSON",
+                    data: data,
+                    xhrFields: {
+                        withCredentials: false
+                    },
+                    success: function(data) {
+                        $('#users').text('');
+                        for(var i=0; i < data.users.length; i++) {
+                            // if(!currentUsers.has(data.users[i])){
+                                currentUsers.add(data.users[i]);
+                                console.log(data.users[i]);
+                                $('#users').append('<div id="userlistName">'+ data.users[i]+'</div>');
+                            }                           
+                        // }                        
+                    }
+                });
+            }
+
+            function showMessages () {
+                $.ajax({
+                    type: 'GET',
+                    url: '/messages',
+                    dataType: "JSON",
+                    data: data,
+                    xhrFields: {
+                        withCredentials: false
+                    },
+                    success: function(data) {
+                        for(var i = 0; i < data.messages.length; i++) {
+                            var timestamp = new Date(parseInt(data.messages[i].timestamp));
+                            var humanTime = timestamp.toUTCString().slice(0, 22);
+                            if (!currentMessages.has(data.messages[i].id)) {
+                                currentMessages.add(data.messages[i].id);
+                                $('#messages').append('<div id="' + data.messages[i].id + '"><div class="messageWrapper"><div class="messageContent">' + '<div class="username">' + data.messages[i].username + ':</div><div class="messageBody" id="messageBody_'  +i + '">' + data.messages[i].body + '</div></div><div class="messageTime">' + humanTime + '</div></div></div>');

```

Comment:
To do osobnej funkcji? Albo z wykorzystaniem ``? W ten sposób to bardzo ciężko czytać. W osobnej funkcji można by zrobić linia pod linią, byłoby widać co tu jest.